graphql/authorizations.ts

```diff
@@ -93,6 +94,12 @@ async function accessCheck(context: GraphQLContext, requiredRoles: Role[], model
             return true;
         }
     }
+    if (requiredRoles.includes(Role.APPOINTMENT_PARTICIPANT)) {
+        assert(modelName === 'Lecture', `Type must be a Lecture to determine access to it`);
+        assert(root, 'root value must be bound to determine access');
+        assert(context.user, 'User must be defined');
+        return isAppointmentParticipant(root, context.user);
```

Comment:
if (isParticipant(...)) {
  return true;
}

Also for the Unauthenticated User this throws an Internal Server Error and should throw an AuthenticationError instead.

Roles that do not match should just be skipped (as only one matching role is required).

Inspired by https://github.com/corona-school/backend/pull/526 ...